From 1ee510baa80c834bbcf77a008e5668dbf3eccf4d Mon Sep 17 00:00:00 2001
From: Alexander Saoutkin <a.saoutkin@gmail.com>
Date: Sat, 2 Jan 2021 20:01:16 +0000
Subject: [PATCH] Check return value of createNodeFromUDSEntry()

createNodeFromUDSEntry() can return a nullptr, which it does when
passed any URL from the baloosearch protocol.

BUG: 431079
---
 kiofusevfs.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/kiofusevfs.cpp b/kiofusevfs.cpp
index 6f275cb..fbf6e27 100644
--- a/kiofusevfs.cpp
+++ b/kiofusevfs.cpp
@@ -393,6 +393,11 @@ void KIOFuseVFS::findAndCreateOrigin(QUrl url, QStringList pathElements, std::fu
 		if(!finalNode)
 		{
 			finalNode = createNodeFromUDSEntry(statJob->statResult(), currentNode->m_stat.st_ino, targetPathComponents.last());
+			if(!finalNode)
+			{
+				qWarning(KIOFUSE_LOG) << "Unable to create a valid final node for" << url << "from its UDS Entry";
+				return callback({}, EIO);
+			}
 			insertNode(finalNode);
 		}
 
-- 
GitLab

